### DISCUSSION TITLE
image당 object 개수 outlier 제거 후 학습

### DISCUSSION THREAD

**Discussion by JiyouSeo** [boostcampaitech2/object-detection-level2-cv-06#113] [ideas]
#11에서 준혁님꼐서 진행하셨던 학습을 outlier들만 제거하여 학습시키는 방법으로 재시도해볼 생각입니다. 
아무래도 object 개수가 많은 경우에 object끼리 더 많이 겹쳐있을 가능성이 높기 때문입니다.
### **사진 0(object가 40개 이상인 이미지들)**
![image](https://user-images.githubusercontent.com/61641072/136646672-bd914087-8db7-42e4-a4d4-0c81b69b9d16.png)
![image](https://user-images.githubusercontent.com/61641072/136646680-9beb6a8a-d899-464b-9520-bef6f0afdb76.png)
![image](https://user-images.githubusercontent.com/61641072/136646688-20c08408-3d22-4a14-aa60-b53c76860c85.png)
![image](https://user-images.githubusercontent.com/61641072/136646693-8978bc4c-43a8-4841-b757-cfe0d35931f5.png)

### **사진 1**
사진 1은 team_train의 각 image당 object 개수의 분포입니다.(object가 1인 image가 1790개)
![image](https://user-images.githubusercontent.com/61641072/136646321-24fe1cc4-2cd2-4beb-aa36-34e6510e13bd.png)
![image](https://user-images.githubusercontent.com/61641072/136646351-d9d85dcf-511f-43dd-a506-13137be99400.png)
그리고 사진1을 기준으로
### **사진 2**
![image](https://user-images.githubusercontent.com/61641072/136646392-e170ae99-e916-483d-bfb3-d61b2fbd3e16.png)
앞의 분포를 box plot으로 그렸을 때, 
제 3사분위에서 1.5 IQR을 더한 위치의 이후인 outlier는 13~14정도입니다.(혹시 제 해석에 틀린 부분이 있으면 말씀해주세요 !)
### **사진 3**
![image](https://user-images.githubusercontent.com/61641072/136646177-8b9875a9-b223-4d33-84d3-da170535d2c3.png)
따라서, object 개수가 14이하인 image들만 json으로 따로 뽑아서 학습을 돌려볼 계획입니다.
사실 사진 1에서 object 개수가 27이하인 image들로 학습을 돌려보고 싶기도 합니다..(14이하로 자르면 train set 개수를 너무 줄이는 건 아닐까 걱정이 되어서요.)
만약 이 방법이 효과가 있으면 만들어낸 json에 mosaic까지 더 해볼 생각입니다 !

**Comment by jaehwan-AI** [comment]
object 개수가 14개 이하인 image들의 annotation 분포 그래프도 그려서 확인해보면 좋을 것 같습니다

**Reply by JiyouSeo** [reply]
![image](https://user-images.githubusercontent.com/61641072/136683981-6008933f-3d03-41a1-95c5-b350f3731920.png)
오오...이렇게 말씀하시는 건가요?!

**Reply by jaehwan-AI** [reply]
아 제가 단어 선택을 실수했네요.. class 분포를 말씀드린 거였습니다!
